tests/vppinputasync.cpp

```diff
@@ -109,4 +109,25 @@ bool VppInputAsync::init(const char* inputFileName, uint32_t fourcc, int width,
 {
     assert(0 && "no need for this");
     return false;
+}
+
+int VppInputAsync::getWidth()
+{
+    if (m_input && m_width != m_input->getWidth())
```

Comment:
But what I concern is that one day in the future, this VppInputAsync::init() is going to be implemented and not to initiate m_input, m_input->getWidth() will raise up a segment error. And the m_input is not the member we have to initiate, because class  VppInputAsync derives from VppInput too, it can override read() function to implement the features which m_input has.